### PR TITLE
general code-cleanup

### PR DESCRIPTION
**Pull request by arogge**
* migrate `host.h` and friends into `config.h`
* remove some unused definitions from `config.h` and their cmake counterparts
* allow usage of `FORTIFY_SOURCE` (which will generate a lot of additional warnings)
* introduce new macro `INTENDED_FALLTHROUGH`
* fix some old and new compiler warnings

When reviewing, looking at the individual commits is probably easiest.

TODO:
* remove even more definitions from `config.h`
* remove all unused checks from `BareosCheckFunctions.cmake`
* maybe move config.h into `CMAKE_BINARY_DIR`

### CODE REVIEW

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -385,8 +385,6 @@ set(LOCALEDIR \"${CMAKE_INSTALL_LOCALEDIR}\")
 
 set(HAVE_CRYPTO 1)
 
-include(BareosTypeSizes)
```
Comment (franku) [inline core/CMakeLists.txt:395]: This whole commit is one of the coolest I have seen on this code. ;-) Very good, now we have stdint.h.

File: core/CMakeLists.txt [hunk 2]
```diff
@@ -745,12 +745,12 @@ message(" ")
...
 message(" ")
 
-message("   Host:                         ${host} -- ${DISTNAME} ${DISTVER} ")
+message("   Host:                         ${host} -- ${PLATFORM} ${DISTVER} ")
 message(
   "   Bareos version:               ${BAREOS} ${BAREOS_FULL_VERSION} (${DATE}) "
 )
 message(
-  "   Distribution:                 ${DISTNAME}"
+  "   Build platform:               ${PLATFORM}"
```
Comment (franku) [inline core/CMakeLists.txt:753]: "Build Host" should also be common.

File: core/src/lib/bsys.cc [hunk 3]
```diff
@@ -217,23 +217,14 @@ char* bstrinlinecpy(char* dest, const char* src)
   if (src <= dest) { return NULL; }
 
   len = strlen(src);
-#if HAVE_BCOPY
```
[the remark this answers is not in the document]
Comment (arogge) [reply core/src/lib/bsys.cc:220]: Das `bstrlinecpy()` bestimmt die Länge des Strings, das macht `memmove()` natürlich nicht.

File: core/CMakeLists.txt (hunk 2 above)
Comment (arogge) [reply core/CMakeLists.txt:753]: I thought this through. For me a "Build Host" is one specific system (i.e. "my laptop"), thus I ended up with "Build platform" which will end up containing "redhat" or "debian" or "solaris".

**Comment by arogge** [comment]
Rebased to latest master and spell-checked.

**Pull request merged**